homeassistant/components/brother/sensor.py

```diff
@@ -0,0 +1,113 @@
+"""Support for the Brother service."""
+import logging
+
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    ATTR_DRUM_COUNTER,
+    ATTR_DRUM_REMAINING_LIFE,
+    ATTR_DRUM_REMAINING_PAGES,
+    ATTR_ICON,
+    ATTR_LABEL,
+    ATTR_MANUFACTURER,
+    ATTR_UNIT,
+    DOMAIN,
+    SENSOR_TYPES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add Brother entities from a config_entry."""
+    brother = hass.data[DOMAIN][config_entry.entry_id]
+
+    sensors = []
+
+    name = brother.model
+    device_info = {
+        "identifiers": {(DOMAIN, brother.serial)},
+        "name": brother.model,
+        "manufacturer": ATTR_MANUFACTURER,
+        "model": brother.model,
+        "sw_version": brother.firmware,
+    }
+
+    for sensor in SENSOR_TYPES:
+        if sensor in brother.data:
+            sensors.append(BrotherPrinterSensor(brother, name, sensor, device_info))
+    async_add_entities(sensors, True)
+
+
+class BrotherPrinterSensor(Entity):
+    """Define an Brother Printer sensor."""
+
+    def __init__(self, data, name, kind, device_info):
```

Comment:
Please rename `data` to `printer` to make it clear that it's a custom instance with attributes and not just a simple container with data.